### ISSUE TITLE
Get reference to default / index file handler

### ISSUE THREAD

**Issue by kaigoh** [BCsabaEngine/svelteesp32#13]
Hi,

First off, amazing work - thank you - you saved some hair pulling dealing with SPIFFS limitations!

To get SvelteKit routing working properly, I need a way of "rewriting" requests to index.html. Is there any way you could extend the C++ output so I can somehow get a reference to the index file handler? The point being I can just do request->send(indexFile) in my server.onNotFound handler.

Or if you have any better ideas?

Thanks :)

**Comment by BCsabaEngine** [comment]
If your dist folder contains any index.htm or html, the default handler (/) will handle it. So you can access website without index.htm postfix in url.

This handler is not exposed, but in next few days I will look forward to implement this.

**Comment by BCsabaEngine** [comment]
If I understand correctly, you should serve not existing routes with index.htm?

**Comment by kaigoh** [comment]
I've added a pull request (https://github.com/BCsabaEngine/svelteesp32/pull/14) that adds some pointers to the default document in the C++ template. There are better ways of doing it, namely a bit of a repetitive rewrite to the template so that the default document has a name other than dataxx, etagxx and funcxx, but this achieves the same goal with minimal other changes.

**Comment by BCsabaEngine** [comment]
Thank you for your first contribution! I have merged and released v1.2.7. (I have made some newline correnction to see routes clearly in header file).

**Please check it in your project with v1.2.7 and cloe issue it you are satisfied!**

**Comment by kaigoh** [comment]
Latest version works really nicely :)

**Comment by BCsabaEngine** [comment]
Sorry, but next 1.3.x version will change with breaking change. I have decided to change to every datat member to named datafield, like this: ```data_index_html```, not only the default. With this you can enforce the existing of needed files.

Example:
```
data_assets_index_B4xhKHXE_css
etag_assets_index_B4xhKHXE_css
...
data_index_html
etag_index_html
...
```
You must change the reference from defaultDocument to data_index_html or data_index_htm.

**Comment by kaigoh** [comment]
Seems a sensible approach - predictable naming all round can only be a good thing 👍

-------- Original Message --------

**Comment by BCsabaEngine** [comment]
We have released v1.3 with some c++ defines too...
https://github.com/BCsabaEngine/svelteesp32/releases/tag/v1.3.0
